Docu for globalsearch (#1432)
SAP/luigi: docs/navigation-parameters-reference.md

  - **title** defines the application title. This is shown in the **appSwitcher** drop-down as well as the title in the header of the Luigi application if a user is in the context of the app.
  - **subTitle** defines the application sub-title. This is shown as the sub-title in the header of the Luigi application if a user is in the context of the app.
  - **link** is a link within the Luigi application that defines the root of the app. It is used to switch to the application if the drop-down entry is selected. It is also used to determine if a user is within the app's scope, so that the corresponding title and sub-title can be rendered in the header.

## Global search

> The global search is an input field available in the top navigation bar. The search is available if the search provider object is configured and implemented in the config file.

Reviewer: ```suggestion
The global search is an input field available in the top navigation bar. The search is available if the search provider object is configured and implemented in the `luigi-config.js` file.
```

### searchProvider
- **type**: Object
- **description**: The search provider is an object which contains different events and the possibility to implement a custom result renderer or change only the search result item.

>   **attributes**

Reviewer: ```suggestion
 - **attributes:**
```

  - **onInput**
    - **type**: Function
    - **description**: will be executed on every key-up event.
  - **onEnter**
    - **type**: Function

>     - **description**: will be executed, when the user presses 'enter'. 

Reviewer: ```suggestion
    - **description**: will be executed when the user presses 'Enter'. 
```

  - **onEscape**
    - **type**: Function

>     - **description**: will be executed, when the user presses 'escape'. 

Reviewer: ```suggestion
    - **description**: will be executed when the user presses 'Escape'. 
```

  - **customSearchResultRenderer** 
    - **type**: Function
    - **description**: This function allows you to append your custom search result to a slot which Luigi provides for you. If this function is implemented the default search result popover is disabled.

>     - **attributes**

Reviewer: ```suggestion
    - **attributes**:
```

      - **searchResults**
        - **type**: Array

>         - **description** array of search result items

Reviewer: ```suggestion
        - **description**: array of search result items
```

      - **slot**
        - **type**: DIV element
        - **description**: `div` element as slot. You can append a custom implementation of the search result to this `div` element.
      - **searchApiObj**
        - **type**: Object

>         - **description** It's an object with a function `fireItemSelected` as property. This function gets a search result item as parameter and fires the search provider event `onSearchResultItemSelected`.

Reviewer: ```suggestion
        - **description**: It is an object with a function `fireItemSelected` as property. This function gets a search result item as parameter and fires the search provider event `onSearchResultItemSelected`.
```

  - **customSearchResultItemRenderer**
    - **type**: Function
    - **description**: This function allows you to customize the single list element rendered in the default search result popover.
    - **attributes**:
      - **searchResultItem**
        - **type**: Object

>         - **description** search result item

Reviewer: ```suggestion
        - **description**: search result item
```

      - **slot**

>         - **type** LI element

Reviewer: ```suggestion
        - **type**: LI element
```

>         - **description** `li` element as slot. You can append a custom implementation of a `searchResultItem` to this `li` element.

Reviewer: ```suggestion
        - **description**: `li` element as slot. You can append a custom implementation of a `searchResultItem` to this `li` element.
```

      - **searchApiObj**
        - **type**: Object

>         - **description** It's an object with a function `fireItemSelected` as property. This function gets a `searchResultItem` as parameter and fires the search provider event `onSearchResultItemSelected`.

Reviewer: ```suggestion
        - **description**: It is an object with a function `fireItemSelected` as property. This function gets a `searchResultItem` as parameter and fires the search provider event `onSearchResultItemSelected`.
```

  - **onSearchResultItemSelected**
    - **type**: Function

>     - **description**: will be executed, when the user clicks on a `searchResultItem`. 

Reviewer: ```suggestion
    - **description**: will be executed when the user clicks on a `searchResultItem`. 
```

>     - **attribute** [searchResultItem](luigi-core-api.md#globalsearch)

Reviewer: ```suggestion
    - **attribute**: 
      - [searchResultItem](luigi-core-api.md#globalsearch)
```
